splitgraph/ingestion/bigquery/__init__.py

```diff
@@ -0,0 +1,177 @@
+import base64
+from copy import deepcopy
+from typing import TYPE_CHECKING, Any, Dict, Optional
+
+from splitgraph.core.types import Credentials, Params, TableInfo
+from splitgraph.hooks.data_source.fdw import ForeignDataWrapperDataSource
+from splitgraph.ingestion.common import build_commandline_help
+
+if TYPE_CHECKING:
+    from splitgraph.engine.postgres.engine import PostgresEngine
+
+
+class BigQueryDataSource(ForeignDataWrapperDataSource):
+    credentials_schema: Dict[str, Any] = {
+        "type": "object",
+        "properties": {
+            "credentials": {
```

Comment:
There's (currently) no point in letting users of the JSONSchema (which is used in form generation) to pass credentials via a path. I think this could be simplified to treat the commandline-passed credential string as a path and the one passed via `__init__` as a JSON-serialized credential.

JSONSchema:

```json
"credentials": {
    "type": "string",
    "title": "GCP credentials",
    "description": "GCP credentials in JSON format",
}
```

commandline: 

```
$ sgr mount bigquery bq -o@- <<EOF
{
    "credentials": "/path/to/my/creds.json",
    "project": "my-project-name",
    "dataset_name": "my_dataset"
}
EOF
```

...

```python
credentials = Credentials({})

with open(params.pop("credentials"), "r") as credentials_file:
    credentials_str = credentials_file.read()

params.pop("credentials")
credentials["credentials"] = credentials_str
```